graylog2-web-interface/src/views/logic/views/CopyWidgetToDashboard.ts

```diff
@@ -21,24 +21,47 @@ import View from 'views/logic/views/View';
 import Query from 'views/logic/queries/Query';
 import GenerateNextPosition from 'views/logic/views/GenerateNextPosition';
 import WidgetPosition from 'views/logic/widgets/WidgetPosition';
+import TitleTypes from 'views/stores/TitleTypes';
 
 import UpdateSearchForWidgets from './UpdateSearchForWidgets';
 import FindWidgetAndQueryIdInView from './FindWidgetAndQueryIdInView';
 
 type QueryId = string;
 
-const _addWidgetToDashboard = (widget: Widget, dashboard: View, oldPosition: WidgetPosition): View => {
-  const dashboardQueryId = dashboard.state.keySeq().first();
-  const viewState = dashboard.state.get(dashboardQueryId);
-  const widgets = viewState.widgets.push(widget);
-  const { widgetPositions } = viewState;
+const _newPositionsMap = (oldPosition, widgetPositions, widget, widgets) => {
   const widgetPositionsMap = oldPosition ? {
     ...widgetPositions,
     [widget.id]: oldPosition.toBuilder().row(0).col(0).build(),
   } : widgetPositions;
-  const newWidgetPositions = GenerateNextPosition(Map(widgetPositionsMap), widgets.toArray());
+
+  return GenerateNextPosition(Map(widgetPositionsMap), widgets.toArray());
+};
+
+const _newTitlesMap = (titlesMap, widget, title) => {
+  if (!title) {
+    return titlesMap;
+  }
+
+  const widgetTitles = titlesMap.get(TitleTypes.Widget);
+  const newWidgetTitles = widgetTitles.set(widget.id, title);
```

Comment:
`widgetTitles` can be `undefined`. It should be initialized with an empty `Immutable.Map` if it is not present.